components/automate-ui/src/app/entities/userperms/userperms.effects.ts

```diff
@@ -17,27 +28,35 @@ import {
   GetUserParamPermsFailure
 } from './userperms.actions';
 import { UserPermsRequests } from './userperms.requests';
+import { getlastFetchTime } from './userperms.selectors';
 
 @Injectable()
 export class UserPermEffects {
   constructor(
     private actions$: Actions,
-    private requests: UserPermsRequests
+    private requests: UserPermsRequests,
+    private store: Store<NgrxStateAtom>
   ) { }
 
-  // include debounce here because each request of this type is a true duplicate
   @Effect()
   fetchAllPerms$ = this.actions$.pipe(
     ofType(UserPermsTypes.GET_ALL),
-    debounceTime(20),
-    mergeMap(() => this.requests.fetchAll().pipe(
-      map((resp: UserPermsResponsePayload) => new GetAllUserPermsSuccess(resp.endpoints)),
-      catchError((error: HttpErrorResponse) => of(new GetAllUserPermsFailure(error))))));
+    debounceTime(20), // each request of this type is a true duplicate so debounce them
+    withLatestFrom(this.store.select(getlastFetchTime)),
+    filter(([_action, lastTime]) => this.stale(lastTime)),
```

Comment:
(4) The cache expiration check--stop if the data is not yet stale; otherwise re-fetch the data.